src/components/features/KeyManagerNonceChecker/KeyManagerNonceChecker.tsx

```diff
@@ -1,52 +1,48 @@
 /* eslint-disable react/no-unescaped-entities */
 
-import { useState } from 'react';
+import { useState, useContext } from 'react';
+import { isAddress, getAddress, type Address } from 'viem';
+import { useReadContract } from 'wagmi';
 
-import LSP6KeyManager from '@lukso/lsp-smart-contracts/artifacts/LSP6KeyManager.json';
-import useWeb3 from '@/hooks/useWeb3';
+import LSP6KeyManager from '@lukso/lsp6-contracts/artifacts/LSP6KeyManager.json';
 import { LSP_SPECS_URL } from '@/constants/links';
+import { NetworkContext } from '@/contexts/NetworksContext';
+import { getChainIdByNetworkName } from '@/config/wagmi';
 
 const KeyManagerNonceChecker: React.FC = () => {
-  const web3 = useWeb3();
+  const { network } = useContext(NetworkContext);
+  const chainId = getChainIdByNetworkName(network.name);
 
   const [keyManagerAddress, setKeyManagerAddress] = useState('');
   const [callerAddress, setCallerAddress] = useState('');
   const [channelId, setChannelId] = useState('');
-  const [nonce, setNonce] = useState('');
+  const [shouldFetch, setShouldFetch] = useState(false);
 
-  const [showNonce, setShowNonce] = useState(false);
-  const [error, showError] = useState(false);
+  const isValidKeyManager = isAddress(keyManagerAddress);
+  const isValidCaller = isAddress(callerAddress);
 
-  const getNonceFromKeyManager = async () => {
-    if (!web3) return;
-
-    const keyManagerInstance = new web3.eth.Contract(
-      LSP6KeyManager.abi as any,
-      web3.utils.toChecksumAddress(keyManagerAddress),
-    );
-
-    const isKeyManagerV05 = await keyManagerInstance.methods
-      // Caution: Fixed Interface ID
-      .supportsInterface('0x6f4df48b')
-      .call();
+  const {
+    data: nonce,
+    isError,
+    isSuccess,
+    refetch,
+  } = useReadContract({
+    address: isValidKeyManager ? getAddress(keyManagerAddress) : undefined,
+    abi: LSP6KeyManager.abi as any,
```

Comment:
you can get this abi from `@lukso/lsp6-contracts`, i think its `lsp6KeyManagerAbi`